examples/open-ai-plugin/packages/service/src/todo.ts

```diff
@@ -0,0 +1,94 @@
+import { extendApi } from "@anatine/zod-openapi";
+import { command, entity } from "@eventual/core";
+import { ulid } from "ulidx";
+import z from "zod";
+
+const userNameSchema = extendApi(z.string(), {
+  description: "The name of the user",
+});
+
+const todoIdSchema = extendApi(z.string(), {
+  description: "Id of a single todo item",
+});
+
+const todoTextSchema = extendApi(z.string(), {
+  description: "The display text of a single todo task",
+});
+
+const todoSchema = z.object({
+  username: userNameSchema,
+  todoId: todoIdSchema,
+  text: todoTextSchema,
+});
+
+const todos = entity("todos", todoSchema);
+
+export const addTodo = command(
+  "addTodo",
+  {
+    summary: "Add a todo to the list",
+    path: "/:username/todos",
+    method: "POST",
+    params: {
+      username: "path",
+    },
+    input: todoSchema.omit({ todoId: true }),
+    output: todoSchema,
+  },
+  async (request) => {
+    const id = ulid();
+
+    const todo: z.infer<typeof todoSchema> = {
```

Comment:
Poor use of infer in example. Extract out to an interface or type next to the schema.